engine/classes/Elgg/Database/AccessCollections.php

```diff
@@ -369,6 +371,10 @@ public function getWhereSql(array $options = array()) {
 			}
 			$clauses_str .= '(' . implode(' AND ', $clauses['ands']) . ')';
 		}
+		
+		if (empty($clauses_str)) {
+			$clauses_str = "$table_alias{$options['access_column']} = " . ACCESS_PUBLIC;
```

Comment:
`1 = 0` is too extreme, it will crash the installation, because neither the site nor the user will be available.